AutoFinance.Broker/InteractiveBrokers/Controllers/TwsControllerBase.cs

```diff
@@ -179,6 +179,88 @@ public Task<ConcurrentDictionary<string, string>> GetAccountDetailsAsync(string
+        }
+
+        /// <summary>
+        /// Get an account detail snapshot
+        /// </summary>
+        /// <param name="accountId">The account Id</param>
+        /// <param name="cancellationToken">The cancellation token used to cancel the request</param>
+        /// <returns>The account values</returns>
+        public async Task<ConcurrentDictionary<string, string>> GetAccountDetailsSnapshotAsync(
+            string accountId, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(accountId))
+                throw new ArgumentException(nameof(accountId));
+
+            // Ensure connection (your class already has EnsureConnectedAsync)
+            await EnsureConnectedAsync();
+
+            // Prevent concurrent snapshots from interfering with each other/clearing cache
+            await _acctSnapLock.WaitAsync(cancellationToken);
+            try
+            {
+                // 1) First cancel old subscription (ignore exceptions when not subscribed)
+                try { this.clientSocket.ReqAccountDetails(false, accountId); } catch { }
+
+                // 2) Clear cache to ensure this is the current data
+                this.accountUpdates.Clear();
+
+                // 3) Bind one-time End handler (bind before sending request to avoid race condition)
+                var tcs = new TaskCompletionSource<ConcurrentDictionary<string, string>>(
+                    TaskCreationOptions.RunContinuationsAsynchronously);
+
+                EventHandler<AccountDownloadEndEventArgs> endHandler = null;
+                // Unbind and cancel task when cancelled
+                var ctr = cancellationToken.Register(() =>
+                {
+                    if (endHandler != null)
+                        this.twsCallbackHandler.AccountDownloadEndEvent -= endHandler;
+                    tcs.TrySetCanceled(cancellationToken);
+                });
+
+                endHandler = (sender, args) =>
+                {
+                    // Unbind after receiving once
+                    this.twsCallbackHandler.AccountDownloadEndEvent -= endHandler;
+                    ctr.Dispose();
+
+                    // (Optional) Cancel subscription again to make this method strictly semantic as "snapshot"
+                    try { this.clientSocket.ReqAccountDetails(false, accountId); } catch { }
+
+                    // Return a copy to avoid caller modifying internal dictionary
+                    var snapshot = new ConcurrentDictionary<string, string>(this.accountUpdates);
+                    tcs.TrySetResult(snapshot);
+                };
+                this.twsCallbackHandler.AccountDownloadEndEvent += endHandler;
+
+                // 4) Initiate this subscription
+                this.clientSocket.ReqAccountDetails(true, accountId);
+
+                // 5) Wait for completion
+                return await tcs.Task.ConfigureAwait(false);
```

Comment:
No need for the configureawait as there is no synchronization context
```suggestion
                return await tcs;
```